recipe/meta.yaml

```
@@ -39,6 +45,19 @@ requirements:
     - pip
     - pytorch
     - pytorch =*={{ torch_proc_type }}*
+    {% if cuda_major >= 12 %}
+    - libcufft-dev  # [linux or win]
+    - libcurand-dev  # [linux or win]
+    - libcublas-dev  # [linux or win]
+    - cuda-nvrtc-dev  # [linux or win]
+    - cuda-nvtx-dev  # [linux or win]
+    - cuda-profiler-api  # [linux or win]
+    - cuda-nvrtc-dev  # [linux or win]
+    - libcusparse-dev  # [linux or win]
+    - libcusolver-dev  # [linux or win]
```

It looks like these libraries are unused by this package. Inferring from [these CI warnings]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=837161&view=logs&j=b7a895b1-8909-5023-4a22-40a4698fe576&t=821caef9-b8fa-5ab7-47a2-7e7cdae7737e&l=835 ) (trimmed one unrelated warning where the `...` is):

```
WARNING (pyg-lib): dso library package conda-forge/linux-64::cuda-nvtx==12.0.76=hcb278e6_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (pyg-lib): dso library package conda-forge/linux-64::libcublas==12.0.1.189=hcb278e6_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
...
WARNING (pyg-lib): dso library package conda-forge/linux-64::libcufft==11.0.0.21=hcb278e6_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (pyg-lib): dso library package conda-forge/linux-64::libcusparse==12.0.0.76=hcb278e6_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (pyg-lib): dso library package conda-forge/linux-64::libcusolver==11.4.2.57=hcb278e6_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (pyg-lib): dso library package conda-forge/linux-64::libcurand==10.3.1.50=hcb278e6_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (pyg-lib): dso library package conda-forge/linux-64::cuda-nvrtc==12.0.76=h59595ed_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

Given this would suggest we drop these

```suggestion
```

Though please let me know if I've missed something

---

Let's try without it but my last two commits were to add libcusparse and libcusolver.

---

Thank for the pointer! 🙏

Maybe that the DSO warning is misleading?

Looking in [this previous CI build]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=837027&view=logs&j=b7a895b1-8909-5023-4a22-40a4698fe576&t=821caef9-b8fa-5ab7-47a2-7e7cdae7737e&l=451 ). Am seeing these lines:

```
  -- USE_CUDNN is set to 0. Compiling without cuDNN support
  -- USE_CUSPARSELT is set to 0. Compiling without cuSPARSELt support
```

Maybe we need to set these config variables somehow and add `cudnn` and `libcusparse-dev` to have them used. Guessing that would clear the DSO warning (in a more preferable way)

---

Even if they are not linked pyg might be loading these dynamically when required, right? I do not know how these warnings evaluate that a library "is not used".

OTOH, even if they are not needed CMake really wants them to be there:

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
  Please set them or make sure they are set and tested correctly in the CMake files:
  CUDA_nvrtc_LIBRARY (ADVANCED)
      linked by target "pyg" in directory /home/conda/feedstock_root/build_artifacts/pyg-lib_1701829131839/work
  LIBNVTOOLSEXT
      linked by target "pyg" in directory /home/conda/feedstock_root/build_artifacts/pyg-lib_1701829131839/work

[...]

  CMake Error at /home/conda/feedstock_root/build_artifacts/pyg-lib_1701829131839/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.10/site-packages/torch/share/cmake/Caffe2/public/cuda.cmake:217 (set_property):
    The link interface of target "caffe2::cublas" contains:

      CUDA::cublas
[...]
```

---

Also, cusparse**lt** is a diffrent package than cusparse AFAIK.
Lets try and add it.

---

Yeah we can try adding these back. Maybe we start with a subset?

---

If we do add them back, think we can drop the selectors. As the Jinja `if` already checks for CUDA 12, these won't be added on any platform without CUDA 12 support

```suggestion
    - libcufft-dev
    - libcurand-dev
    - libcublas-dev
    - cuda-nvrtc-dev
    - cuda-nvtx-dev
    - cuda-profiler-api
    - cuda-nvrtc-dev
    - libcusparse-dev
    - libcusolver-dev
```